### PR TITLE
Project service

### PR DESCRIPTION
**Pull request by majst01**
This enables a user to fetch all projects visible for him, required to make the api usage more useful.

Depends on #127

### CODE REVIEW

File: proto/api/v1/project.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+package api.v1;
+
+import "api/v1/common.proto";
+
+// ProjectService serves project related functions
+service ProjectService {
+  // List all accessible projects
+  rpc List(ProjectServiceListRequest) returns (ProjectServiceListResponse) {
+    option (visibility) = VISIBILITY_SELF;
```
Comment (Gerrit91) [inline proto/api/v1/project.proto:11]: Can also be excluded from auditing.

File: proto/api/v1/project.proto [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+// Project defines a group of resources belonging to a tenant
+// a tenant can have multiple projects
+message Project {
+  // Uuid of this project
+  string uuid = 1;
+  // Name of this project
+  string name = 2;
+  // Tenant this project belongs to
+  string tenant = 3;
```
Comment (Gerrit91) [inline proto/api/v1/project.proto:23]: Description and creation / update timestamp should be added, too.

File: proto/api/v1/project.proto [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+package api.v1;
+
+import "api/v1/common.proto";
+
+// ProjectService serves project related functions
+service ProjectService {
+  // List all accessible projects
+  rpc List(ProjectServiceListRequest) returns (ProjectServiceListResponse) {
+    option (visibility) = VISIBILITY_SELF;
+  }
```
Comment (Gerrit91) [inline proto/api/v1/project.proto:12]: I think `Get` should also be added for the sake of completeness, e.g. when the frontend wants to retrieve project information just for the active project or something.

```
  rpc Get(ProjectServiceGetRequest) returns (ProjectServiceGetResponse) {
    option (project_roles) = PROJECT_ROLE_OWNER;
    option (project_roles) = PROJECT_ROLE_EDITOR;
    option (project_roles) = PROJECT_ROLE_VIEWER;
    option (auditing) = AUDITING_EXCLUDED;
  }
```

**Review by Gerrit91 — COMMENTED** [review]
I also started with this already but did not push before vacation. 🙈

Added some comments about what my changes did include.

File: proto/api/v1/project.proto (hunk 1 above)
Comment (majst01) [reply proto/api/v1/project.proto:13]: done

File: proto/api/v1/project.proto (hunk 2 above)
Comment (majst01) [reply proto/api/v1/project.proto]: done

File: proto/api/v1/project.proto (hunk 3 above)
Comment (majst01) [reply proto/api/v1/project.proto:22]: done

File: proto/api/v1/project.proto [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  // Uuid of this project
+  string uuid = 1;
+  // Name of this project
+  string name = 2;
+  // Description of this project
+  string description = 3;
+  // Tenant this project belongs to
+  string tenant = 4;
+  // CreatedAt the date when this project was created
+  google.protobuf.Timestamp created_at = 10;
```
Comment (Gerrit91) [inline proto/api/v1/project.proto:37]: Is numbering starting at 10 for timestamps intended?

Comment (majst01) [reply proto/api/v1/project.proto:37]: yes to leave room for more properties, created and updated times are always kept at the end

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**